src/service/resources-impl.js

```diff
@@ -1493,6 +1518,8 @@ export class Resources {
       });
       this.requestsChangeSize_ = this.requestsChangeSize_.filter(
           request => request.resource != resource);
+      this.pendingBuildResources_ = this.pendingBuildResources_.filter(
```

Comment:
Thanks for clarifying this on chat, correct, this needs to happen when the element is removed not unloaded. Also during chat we discussed the extra fix to cleanup layouting tasks for a resource when it's removed as well as unloaded.